protocols/servlet/src/test/java/org/jboss/arquillian/protocol/servlet/ProtocolTestCase.java

```diff
@@ -50,6 +50,27 @@ public void shouldReturnTestResult() throws Exception {
             result.getThrowable());
     }
 
+    @Test
+    public void shouldUseLogicalMethodNameAndHandleUrlEscaping() throws Exception {
```

Comment:
This test is actually testing two things - using logical names for test methods and url escaping of test method names in `ServletMethodExecutor`. I was unsure if this should be therefore split into two test or if that would be deemed an overkill. Please let me know what you think.

Reply:
There are two things that I find not great about explicitly testing that the name is url encoded:
* the URL encoding is currently buried deep in `ServletMethodExecutor.invoke()` so to be able to test it we would need to extract it out from there and I find that quite significant a effort which I'd argue is out of scope of this PR,
* as an Arquillian user I want to verify that if a test method has spaces it will still execute in the container as expected and the fact that it gets URL encoded along the way is more of an implementation detail.

In the lights of the above I'm thinking about changing the name of the test in question to `shouldUseLogicalMethodNameAndSupportUrlReservedCharactersInIt`. What are your views on that?

Reply:
I agree that doing an explicit check can result in scope creep for this PR, hence to do it right we would need to refactor the whole thing. I think the rename is giving a bit of a hint for the intent of the test, so that is fine with me.